### PR TITLE
feature/OE-217-region-specific-folder

### PR DESCRIPTION
**Pull request by devsquad-mathews-lima**
## Review

For the submitter, initial self-review:

- [ ] Reviewed the test cases added for completeness and possible points for discussion
- [ ] Check the scope of the changeset - is everything in the diff required for the pull request?

For the pull request reviewer(s), this changeset has been reviewed for:

- [ ] Consistency between the changeset and the goal stated above
- [ ] Usage friction - is the proposed change in usage cumbersome or complicated?
- [ ] Complexity - is the proposed change introducing any unjustified complexity? 
- [ ] Thoroughness of added tests and any missing edge cases

### CODE REVIEW

File: app/Http/Livewire/ShowTrainings.php [hunk 1]
```diff
@@ -79,23 +79,26 @@ public function changeDepartment()
...
-        return $trainingsQuery->get();
+        return TrainingPageSection::query()
+            ->with('content')
+            ->where(function (Builder $query) {
+                $query
+                    ->where('department_id', $this->department->id)
+                    ->orWhereNull('region_id')
+                    ->whereHas('region.offices', function (Builder $query) {
+                        $query->where('offices.id', user()->office_id);
+                    });
```
Comment (Ron-Junior) [inline app/Http/Livewire/ShowTrainings.php]: I'm thinking. maybe this orWhereNull never will be executed, because the whereHas determines that the regions exist.

**Review by Ron-Junior — APPROVED** [review]
